usecases/schema/add_property.go

```diff
@@ -84,9 +87,31 @@ func (m *Manager) addClassProperty(ctx context.Context,
 	return m.addClassPropertyApplyChanges(ctx, className, prop)
 }
 
-func (m *Manager) setNewPropDefaults(class *models.Class, prop *models.Property) {
+func (m *Manager) setNewPropDefaults(class *models.Class, prop *models.Property) error {
 	m.setPropertyDefaults(prop)
+	if err := validateUserProp(class, prop); err != nil {
+		return err
+	}
 	m.moduleConfig.SetSinglePropertyDefaults(class, prop)
+	return nil
+}
+
+func validateUserProp(class *models.Class, prop *models.Property) error {
+	if prop.ModuleConfig != nil {
```

Comment:
return early please eg

```
if prop.ModuleConfig == nil {
    return nil
}
```

Reply:
with the current code also if the `prop.ModuleConfig == nil` then it won't enter the if condition and directly return nil. Not sure how this would help. Let me know if I'm missing something.

Reply:
nothing is wrong with how it is currently done, it's more a it's more a coding style issue. :)

"Catching" some cases early and returning reduces the complexity and mental load of understanding what a function does. But this function is so simple that it doesn't really matter, so feel free to ignore if you prefer the current way

Reply:
Makes sense. No there is nothing like what I prefer. I was just trying to check if there is anything that I missed.

Reply:
I'm a huge fan of [keep code left](https://gist.github.com/ZacharyRSmith/343962433a89ef1b454a/) and really think it improves things. But ofc everybody has different opinions :)